src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1584,6 +1604,7 @@ def translate_device(dev):
             elif not await self.disk_uniqueness_integrity_check(device, vm_instance):
                 verrors.add('attributes.path', f'{vm_instance["name"]} has "{path}" already configured')
         elif device.get('dtype') == 'NIC':
+            await self.nic_check(device)
```

Comment:
Shouldn't this be done after you run basic checks that can raise validation errors?